0.21.10 release blogpost (#252)
TeamNewPipe/website: _posts/2021-09-20-newpipe-0.21.10-released.md

---
layout: post
title: "NewPipe 0.21.10 released: Here's your order! Enjoy the Full Screen!"
short: "NewPipe 0.21.10 released"
date: 2021-09-10 17:00:00 + 1:00
categories: [pinned, release]
author:  <a href="https://github.com/opusforlife2">@Opusforlife</a> & <a href="https://github.com/poolitzer">@Poolitzer</a>
image: newpipe
excerpt_separator: <!-- more -->
---

This release brings you working imports from your YouTube subscriptions export, an "always go into full screen" toggle when you play a video via the main player, a new Invidious instance, and so much more that you basically have to read on.

<!-- more -->

## New

- Rejoice! [@talanc](https://github.com/talanc) fixed an issue plaguing users since late July: Google changed the way it exports subscriptions from YouTube, rendering the previous import into NewPipe non-functional. You can now import your downloaded [CSV](https://en.wikipedia.org/wiki/Comma-separated_values) file, or even the ZIP file that contains that CSV, directly into NewPipe!

- The behaviour of long press on the player queue was streamlined by [@ktprograms](https://github.com/ktprograms): It now also displays the popup menu from the main player.

> - A cool new setting was added by [@Stypox](https://github.com/Stypox): You can now automatically have videos playing full screen if you use the main player.

Reviewer: ```suggestion
- A cool new setting was added by [@Stypox](https://github.com/Stypox): You can now automatically have videos playing in full screen if you use the main player.
```

- [@Stypox](https://github.com/Stypox) also split search suggestions into two categories: local and remote. Now you have the option to enable or disable them independently of each other.

- [@magicbrothers](https://github.com/magicbrothers) added a new Invidious instance, Tubus, meant for educational videos for children.

> - [@ktprograms](https://github.com/ktprograms) finished the ongoing work of adding the `Show channel details` option where it was still missing, that is, in local playlists, feed items, and history items.

Reviewer: Probably better if double quotes were used instead, as this would be shown as code (i.e. `Show channel details`):
```suggestion
- [@ktprograms](https://github.com/ktprograms) finished the ongoing work of adding the "Show channel details" option where it was still missing, that is, in local playlists, feed items, and history items.
```

### Improved

- [@sauravrao637](https://github.com/sauravrao637) modified the app colours to be a bit darker when the dark theme is selected in the app, which looks loads better to us! He further added a darker splash screen variant when the OS's theme is set to dark. We hope you like the new look too!

- [@Stypox](https://github.com/Stypox) replaced the outdated Android-Universal-Image-Loader library (which has not had any new release in the last 6 years) with the more stable and updated Picasso, a library to download and cache images on Android (think all the thumbnails you see in the app). This fixes a nasty issue where the wrong notification thumbnail size could crash the whole System UI or even forcibly reboot the device! It also makes loading cached images instantaneous, and loading non-cached images smoother.

> - [@sherlockbeard](https://github.com/sherlockbeard) changed the 'video not available' image from a static PNG to a vector XML image, allowing it to scale properly for screens of all sizes.

Reviewer: Probably settling on one quote type is better:
```suggestion
- [@sherlockbeard](https://github.com/sherlockbeard) changed the "video not available" image from a static PNG to a vector XML image, allowing it to scale properly for screens of all sizes.
```

- [@Stypox](https://github.com/Stypox) added the mime-type field to all calls to the SAF file picker, which causes files with the incorrect file format to be greyed out and become non-selectable. This will ensure the user can only choose a ZIP file when importing their database, for example. This should be a great quality-of-life improvement for many users!

### Fixed

> - [@litetex](https://github.com/litetex) together with [@Stypox](https://github.com/Stypox) improved the comments parsing for timestamps, which could have resulted in a crash if it happened in the wrong place at the wrong time

Reviewer: ```suggestion
- [@litetex](https://github.com/litetex), together with [@Stypox](https://github.com/Stypox), improved the comments parsing for timestamps, which could have resulted in a crash if it happened in the wrong place at the wrong time.
```

- [@Aga-C](https://github.com/Aga-C) added line wrapping to the settings texts, so they don't get truncated anymore.

- [@acti0](https://github.com/acti0) fixed a regression from a past release where the name of the video wasn't being shared when sharing the video URL.

- [@litetex](https://github.com/litetex) fixed a bug where you had to tap the Replay button twice for it to actually work.

> - While [@litetex](https://github.com/litetex) fixed the Replay button, they also fixed the direction buttons on DPads for Android TV users: the seekbar now jumps only 10 seconds, instead of several minutes.

Reviewer: ```suggestion
- While [@litetex](https://github.com/litetex) fixed the Replay button, they also fixed the direction buttons on D-Pads for Android TV users: the seekbar now jumps only 10 seconds, instead of several minutes.
```

- [@litetex](https://github.com/litetex), being on a roll, reworked the error panel shown when an error happens. This results in a clearer error display.

- [@Redirion](https://github.com/Redirion) implemented proper closure of audio streams when playback is stopped. This fixed a longstanding bug where NewPipe would create more and more audio sessions but use only one. This messed with equaliser apps.

- [@TobiGr](https://github.com/tobigr) added proper error handling which might occur when trying to play a video in popup mode or get a search suggestion.

> - [@TobiGr](https://github.com/tobigr) also took care of a small UI bug: when one played something in the miniplayer and changed the visibility of the description tab in the meantime, the visibility of the comments tab was changed instead. The app isn't confused anymore, we cannot say the same about the dev...

Reviewer: Is "miniplayer" here referring to the popup player?

Reviewer 2: ```suggestion
- [@TobiGr](https://github.com/tobigr) also took care of a small UI bug: when one played something in the minimized player and changed the visibility of the description tab in the meantime, the visibility of the comments tab was changed instead. The app isn't confused anymore, we cannot say the same about the dev...
```

minimized

Reviewer: Thanks, that sounds clearer to me.

## Translation

- [@Stypox](https://github.com/Stypox) switched to using the custom cancel string everywhere, which is translated to more languages than the default android cancel string.

- They jumped onto that plane and replaced a couple more strings to better translated ones: "ok", "yes", "no", and "cancel". They also removed "add", and renamed "finish" to "ok", because it clearly means "ok", not "finish"!

- [@Aga-C](https://github.com/Aga-C) added plural forms for download-related strings.

> - [@CBSkarmory](https://github.com/CBSkarmory) improved some wording in the issue template, thanks!

Reviewer: Yeah, I know, pedants...
```suggestion
- [@CBSkarmory](https://github.com/CBSkarmory) improved some wording in the issue template; thanks!
```

## Nerd Talk

- [@TobiGr](https://github.com/TobiGr) updated the hardcoded SoundCloud client ID.

- [@FireMasterK](https://github.com/FireMasterK) added support for extracting the channel avatar on YouTube and PeerTube.

- [@FireMasterK](https://github.com/FireMasterK) also added the `uploaderVerified` info to the `StreamInfo` class.

- [@litetex](https://github.com/litetex) switched GitHub Actions to using the Eclipse Temurin Java distribution, as AdoptOpenJDK is getting deprecated.

- [@Redirion](https://github.com/Redirion) replaced NewPipe’s custom buffer values with ExoPlayer's default ones.

- [@Isira-Seneviratne](https://github.com/Isira-Seneviratne) changed `PackageInfo` to its backwards-compatible version, `PackageInfoCompat` to fetch the app signatures.

- [@litetex](https://github.com/litetex) also updated the version of Kotlin used everywhere, and fixed some build problems.

- They also introduced a cache setting in the CI configuration which lets it run faster.

> - [@XiangRongLin](https://github.com/XiangRongLin) added a gradle parameter to prevent KtLint, the Kotlin [linter](https://en.wikipedia.org/wiki/Lint_(software))+[formatter](https://en.wikipedia.org/wiki/Prettyprint#Programming_code_formatting), from being run twice in the GitHub Actions [CI](https://en.wikipedia.org/wiki/Continuous_integration).

Reviewer: ```suggestion
- [@XiangRongLin](https://github.com/XiangRongLin) added a Gradle parameter to prevent KtLint, the Kotlin [linter](https://en.wikipedia.org/wiki/Lint_(software))+[formatter](https://en.wikipedia.org/wiki/Prettyprint#Programming_code_formatting), from being run twice in the GitHub Actions [CI](https://en.wikipedia.org/wiki/Continuous_integration).
```

Also, CI is mentioned in the previous entry, so I think that is where the Wikipedia article on CI should be linked.
